src/ui/views/Send/SendToCadence/TransferConfirmation.tsx

```diff
@@ -72,150 +76,158 @@ const TransferConfirmation = (props: TransferConfirmationProps) => {
-          `You have sent ${props.data.amount} ${props.data.tokenSymbol} to ${props.data.contact.contact_name}. \nClick to view this transaction.`,
-          props.data.coinInfo.icon
-        );
-        props.handleCloseIconClicked();
-        await wallet.setDashIndex(0);
-        setSending(false);
-        setTid(txId);
-        history.push(`/dashboard?activity=1&txId=${txId}`);
-      })
-      .catch((err) => {
-        console.log('0xe8264050e6f51923 ', err);
-        setSending(false);
-        setFailed(true);
-      });
-  };
-
-  const withDrawEvm = async () => {
-    console.log('transferToken ->', props.data);
-    setSending(true);
-    if (props.data.tokenSymbol.toLowerCase() === 'flow') {
-      transferFlow();
-    } else {
-      transferFt();
+      );
+      await usewallet.setRecent(props.data.contact);
+      usewallet.listenTransaction(
+        txId,
+        true,
+        `${props.data.amount} ${props.data.coinInfo.coin} Sent`,
+        `You have sent ${props.data.amount} ${props.data.tokenSymbol} to ${props.data.contact.contact_name}. \nClick to view this transaction.`,
+        props.data.coinInfo.icon
+      );
+      props.handleCloseIconClicked();
+      await usewallet.setDashIndex(0);
+      setSending(false);
+      setTid(txId);
+      history.push(`/dashboard?activity=1&txId=${txId}`);
+    } catch (err) {
+      console.log('0xe8264050e6f51923 ', err);
+      setSending(false);
+      setFailed(true);
     }
-  };
-
-  const transferFlow = async () => {
-    const amount = parseFloat(props.data.amount).toFixed(8);
-    // const txID = await wallet.transferTokens(props.data.tokenSymbol, props.data.contact.address, amount);
-    wallet
-      .withdrawFlowEvm(amount, props.data.contact.address)
-      .then(async (txId) => {
-        await wallet.setRecent(props.data.contact);
-        wallet.listenTransaction(
-          txId,
-          true,
-          `${props.data.amount} ${props.data.coinInfo.coin} Sent`,
-          `You have sent ${props.data.amount} ${props.data.tokenSymbol} to ${props.data.contact.contact_name}. \nClick to view this transaction.`,
-          props.data.coinInfo.icon
-        );
-        props.handleCloseIconClicked();
-        await wallet.setDashIndex(0);
-        setSending(false);
-        setTid(txId);
-        history.push(`/dashboard?activity=1&txId=${txId}`);
-      })
-      .catch(() => {
-        setSending(false);
-        setFailed(true);
-      });
-  };
+  }, [props, usewallet, history, setSending, setFailed, setTid]);
 
-  const transferFt = async () => {
-    const tokenResult = await wallet.openapi.getEvmTokenInfo(props.data.tokenSymbol);
-    console.log('tokenResult ', tokenResult, props.data.amount);
+  const flowFromEvm = useCallback(async () => {
+    try {
+      const txId = await usewallet.withdrawFlowEvm(props.data.amount, props.data.contact.address);
+      await usewallet.setRecent(props.data.contact);
```

Comment:
There is quite a lot of duplicated code in these functions. You could write a `handleTransaction` function or something like that:
```suggestion
  const handleTransaction = useCallback(async (txId: string) => {
    await usewallet.setRecent(props.data.contact);
    wallet.listenTransaction(
        txId,
        true,
        `${amount} ${props.data.coinInfo.coin} Sent`,
        `You have sent ${amount} ${props.data.tokenSymbol} to ${props.data.contact.contact_name}. \nClick to view this transaction.`,
        props.data.coinInfo.icon
      );
      props.handleCloseIconClicked();
      await usewallet.setDashIndex(0);
      setSending(false);
      setTid(txId);
      history.push(`/dashboard?activity=1&txId=${txId}`);
  },[wallet,history]
```